File: P02 - Kelas Objek/Java/Bag. kedua/Sepeda.java
package Praktikum.P02;
import java.util.Scanner;

public class Sepeda {
    public String merk;
    public String tipe;
    public int gigi_sekarang;
    public int kecepatan;
    
    public Sepeda(int gigi_sekarang, int kecepatan){
        this.gigi_sekarang = gigi_sekarang;
        this.kecepatan = kecepatan;
    }

    public void setMerk(String merk){
        this.merk = merk;
    }

    public void setTipe(String tipe){
        this.tipe = tipe;
    }

    public void naikkanGigi(){
        this.gigi_sekarang += 1;
    }

    public void turunkanGigi(){
        this.gigi_sekarang -= 1;
    }

    public void tambahKecepatan(int tambah){
        this.kecepatan += tambah;
    }

    public void rem(int kurangi){
        this.kecepatan -= kurangi;
    }

    public void tampilkan(){
        System.out.printf("\nSuatu hari kamu sedang menaiki sepeda dengan merk %s dengan tipe %s \n",  this.merk, this.tipe);
    }

    public static void main(String[] args) {
        Sepeda s1 = new Sepeda(2, 30);
        Scanner sc = new Scanner(System.in);
        
        String merk, tipe, pil;
        int kec;
        Boolean cond = false;
              
        System.out.print("Merk sepeda anda   : ");
        merk = sc.nextLine();
        s1.setMerk(merk);

        System.out.print("Tipe Sepeda        : ");
        tipe = sc.nextLine();
        s1.setTipe(tipe);


        s1.tampilkan();
        System.out.printf(",kamu sedang melaju dengan kecepatan %s km/jam dengan gigi %s\n",  s1.kecepatan, s1.gigi_sekarang);
        System.out.println("Apa yang ingin kamu lakukan? : \n");

        while (!cond) {
            System.out.print("Naikan Kecepatan (Y) / Rem (N)  : ");
            pil = sc.next();
        
            if(pil.equals("Y")){
                System.out.print("Naikan Kecepatan : ");
                kec = sc.nextInt();
                s1.tambahKecepatan(kec);
            }
            else{
                System.out.print("Rem : ");
                kec = sc.nextInt();
                s1.rem(kec);
            }

            if(s1.kecepatan == 0){
                cond = true;
            }
            else{
                System.out.printf("\nKecepatan sekarang : %s km/jam dengan Gigi %s  \n",  s1.kecepatan, s1.gigi_sekarang);
                System.out.print("Naikan Gigi (Y) / Turun (N)  : ");
                pil = sc.next();

                if(pil.equals("Y")){
                    s1.naikkanGigi();
                }
                else{
                    s1.turunkanGigi();
                }

                System.out.printf("\nKecepatan sekarang : %s km/jam dengan Gigi %s  \n",  s1.kecepatan, s1.gigi_sekarang);
            }
            System.out.println("Sepedamu sudah berhenti");    
        }
        
        sc.close();
    }
    
    
           
    
}

File: P02 - Kelas Objek/Python/Bag. kedua/Sepeda.py
# File: Sepeda.py
# Penulis : Elmosius Suli
# Membuat kelas Sepeda

class Sepeda:
    def __init__(self, merk: str, tipe: str, gigi_sekarang: int, kecepatan: int) -> None:
        self.merk = merk
        self.tipe = tipe
        self.gigi_sekarang = gigi_sekarang
        self.kecepatan = kecepatan
        return

    def setMerk(self, merk: str) -> None:
        self.merk = merk

    def setTipe(self, tipe: str) -> None:
        self.tipe = tipe

    def naikkanGigi(self) -> None:
        self.gigi_sekarang += 1

    def turunkanGigi(self) -> None:
        self.gigi_sekarang -= 1

    def tambahKecepatan(self, tambah: int) -> None:
        self.kecepatan += tambah

    def rem(self, kurangi: int) -> None:
        self.kecepatan -= kurangi

    def cetakKondisi(self) -> None:
        print("--------------------------------------------------")
        print(
            f"Kamu sedang mengendarai sepeda merk {self.merk} dengan tipe {self.tipe}")
        print(
            f"Sepeda melaju dengan kecepatan {self.kecepatan} km/jam menggunakan gigi {self.gigi_sekarang}")

## Program utama ##
def main():
    s1 = Sepeda("Giant", "sepeda lipat", 1, 15)

    s1.cetakKondisi()
    print("Apa yang ingin kamu lakukan sekarang?")
    print("1. Mengganti sepeda")
    print("2. Menaikkan gigi")
    print("3. Menurunkan gigi")
    print("4. Menambah kecepatan")
    print("5. Rem")
    print("0. Berhenti")
    pilih = int(input("Silakan masukkan angka : "))

    while (pilih != 0):
        if (pilih == 1):
            merkBaru = input(
                "Merk Sepeda Baru (isikan '-' jika merk tidak berubah) : ")
            tipeBaru = input("Tipe Sepeda Baru : ")
            if (merkBaru != "-"):
                s1.setMerk(merkBaru)
            s1.setTipe()
        elif (pilih == 2):
            s1.naikkanGigi()
        elif (pilih == 3):
            s1.turunkanGigi()
        elif (pilih == 4):
            tambah = int(input("Berapa kecepatan yang ditambah? "))
            s1.tambahKecepatan(tambah)
        elif (pilih == 5):
            kurangi = int(input("Berapa kecepatan yang dikurangi? "))
            s1.rem(kurangi)

        if (s1.kecepatan != 0):
            s1.cetakKondisi()
            print("Apa yang ingin kamu lakukan sekarang?")
            print("1. Mengganti sepeda")
            print("2. Menaikkan gigi")
            print("3. Menurunkan gigi")
            print("4. Menambah kecepatan")
            print("5. Rem")
            print("0. Berhenti")
            pilih = int(input("Silakan masukkan angka : "))

    print("Sepeda kamu telah berhenti")

if __name__ == '__main__':
    main()
